vars/commonPipelineEnvironment.groovy

```diff
@@ -192,36 +192,31 @@ class commonPipelineEnvironment implements Serializable {
     ]
 
     void writeToDisk(script) {
-
         files.each({f  ->
-            if (this[f.property] && !script.fileExists(f.filename)) {
-                script.writeFile file: f.filename, text: this[f.property]
-            }
+            writeValueToFile(script, f.filename, this[f.property])
         })
 
         containerProperties.each({key, value ->
-            def fileName = ".pipeline/commonPipelineEnvironment/container/${key}"
-            if (value && !script.fileExists(fileName)) {
-                if(value in CharSequence) {
-                    script.writeFile file: fileName, text: value
-                } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
-                }
-            }
+            writeValueToFile(script, ".pipeline/commonPipelineEnvironment/container/${key}", value)
         })
 
         valueMap.each({key, value ->
-            def fileName = ".pipeline/commonPipelineEnvironment/custom/${key}"
-            if (value && !script.fileExists(fileName)) {
-                if(value in CharSequence) {
-                    script.writeFile file: fileName, text: value
-                } else {
-                    script.writeFile file: fileName, text: groovy.json.JsonOutput.toJson(value)
-                }
-            }
+            writeValueToFile(script, ".pipeline/commonPipelineEnvironment/custom/${key}", value)
         })
     }
 
+    void writeValueToFile(script, String filename, value){
+        if (value == null)
```

Comment:
[minor][style] I slightly dislike in general the approach of calling a writeValue and check inside if the value is `null` and return immediately. In that case it might be better to have the `null` check outside before calling the method. But on the other hand I see that with that we can avoid the `null` checks elsewhere in the coding.

Reply:
Actually this I changed from `!value` to `value == null`. Any objection on this?

Reply:
> Actually this I changed

With the old approach the empty string was not serialized. Now the empty string is serialized. Can't judge how that affects the existing code base.

Reply:
I changed it back..